### PR TITLE
support custom image size in compiler

### PR DESCRIPTION
**Pull request by pelikhan**
Users have been asking how to create a microbit dot block with a different number of dots. This is a small PR to add number of columns/rows options. It's kind of annoying that we still have this string hack in the way but well...

Test program:
```
namespace custom {
    //% block="show leds"
    //% imageLiteral=1 imageLiteralColumns=7 imageLiteralRows=2
    export function showLeds(leds: string) {
        const img = <Image><any>leds;
        img.plotImage()
    }
}
```
leads to 

image

```
custom.showLeds(`
    # . # # . # .
    . # . # # . #
    `)
```

### CODE REVIEW

File: pxtblocks/blocklyloader.ts [hunk 1]
```diff
@@ -600,8 +600,10 @@ namespace pxt.blocks {
             }
         });
         if (fn.attributes.imageLiteral) {
+            const columns = fn.attributes.imageLiteralColumns || fn.attributes.imageLiteral * 5;
```
Comment (mmoskal) [inline pxtblocks/blocklyloader.ts]:
```suggestion
            const columns = (fn.attributes.imageLiteralColumns || 5) * fn.attributes.imageLiteral;
```

File: pxtcompiler/emitter/decompiler.ts [hunk 2]
```diff
@@ -1336,13 +1336,14 @@ ${output}</xml>`;
             res.fields = [];
 
             const leds = ((arg as ts.StringLiteral).text || '').replace(/\s+/g, '');
-            const nc = attributes.imageLiteral * 5;
-            if (nc * 5 != leds.length) {
+            const nc = attributes.imageLiteralColumns || attributes.imageLiteral * 5;
```
Comment (mmoskal) [inline pxtcompiler/emitter/decompiler.ts]:
```suggestion
            const nc = (attributes.imageLiteralColumns || 5) * attributes.imageLiteral;
```

File: pxtcompiler/emitter/decompiler.ts [hunk 3]
```diff
@@ -2506,8 +2507,9 @@ ${output}</xml>`;
                     return Util.lf("Only string literals supported for image literals")
                 }
                 const leds = ((arg as ts.StringLiteral).text || '').replace(/\s+/g, '');
-                const nc = attributes.imageLiteral * 5;
-                if (nc * 5 != leds.length) {
+                const nr = attributes.imageLiteralRows || 5;
+                const nc = attributes.imageLiteralColumns || attributes.imageLiteral * 5;
```
Comment (mmoskal) [inline pxtcompiler/emitter/decompiler.ts]:
```suggestion
                const nc = (attributes.imageLiteralColumns || 5) * attributes.imageLiteral;
```

**Comment by riknoll** [comment]
Is there a reason this is a draft PR?

**Comment by pelikhan** [comment]
I hadn’t had the time to check if I could add tests for it and I’m on the move.  But it’s mostly ready.

**Comment by pelikhan** [comment]
@riknoll @mmoskal ready

**Pull request merged**